controller/api/public/stat_summary.go

```diff
@@ -585,24 +620,32 @@ func processPrometheusMetrics(req *pb.StatSummaryRequest, results []promResult,
 				addTCPStats()
 				tcpStats[resource].WriteBytesTotal = value
 			}
-
 		}
 	}
 
 	return basicStats, tcpStats
 }
 
-func metricToKey(req *pb.StatSummaryRequest, metric model.Metric, groupBy model.LabelNames) rKey {
-	// this key is used to match the metric stats we queried from prometheus
-	// with the k8s object stats we queried from k8s
-	// ASSUMPTION: this code assumes that groupBy is always ordered (..., namespace, name)
-	key := rKey{
-		Type: req.GetSelector().GetResource().GetType(),
-		Name: string(metric[groupBy[len(groupBy)-1]]),
+// metricToKey generates a key which is used to match the metric stats we
+// queried from prometheus with the k8s object stats we queried from k8s
+func metricToKey(kind string, outboundFrom bool, metric model.Metric) rKey {
+	kindLabel := model.LabelName(k8s.KindToStatsLabel(kind, outboundFrom))
+	namespaceLabel := model.LabelName(k8s.KindToStatsLabel(k8s.Namespace, outboundFrom))
+
+	if _, exists := metric[kindLabel]; !exists {
+		log.Debugf(debugTxtMismatchLabel, "label", kindLabel, metric)
```

Comment:
I made this `Debugf` because if there is a mismatch between the keys used in the k8s stats and Prometheus stats maps, it's more than likely a programming error. There is nothing a user can do about it, but to file GH issue.

Reply:
From what I can tell, the tcp family of metrics (`tcp_read_bytes`, `tcp_write_bytes`, `tcp_open_connections`) don't have `dst_*` labels, so they raise these errors. Maybe they deserve special treatment here.

Reply:
Thanks for pointing this out. These two checks don't make sense because, e.g.:

```
# given this query
sum(increase(tcp_read_bytes_total{direction="inbound", namespace="emojivoto"}[1m])) by (namespace, daemonset, statefulset, k8s_job, deployment, replicationcontroller, pod, service)

# ... the response looks like

Element | Value
------- | --
{k8s_job="voting-job",namespace="emojivoto",pod="voting-job-jj5ms"}       | 7323.6
{k8s_job="web-job",namespace="emojivoto",pod="web-job-fpn8z"}             | 283312.8
{deployment="emoji",namespace="emojivoto",pod="emoji-697b575bd9-rctbl"}   | 145273.2
{deployment="voting",namespace="emojivoto",pod="voting-6b4bf7494b-m7767"} | 7274.4
{deployment="web",namespace="emojivoto",pod="web-559684dbc5-8v45z"}       | 283203.6
{k8s_job="emoji-job",namespace="emojivoto",pod="emoji-job-rjzjk"}         | 142485.59999999998
```
So in the case of `k8s.All`, this function will be called by the `processPrometheusMetrics()` function as it iterates through the different kinds with `metric` that belongs to other workload kinds. I'm removing these `Debugf` statements.